### DISCUSSION TITLE
Firecracker Customer Survey

### DISCUSSION THREAD

**Discussion by xmarcalx** [firecracker-microvm/firecracker#2945] [General]
Dear Community,

Thanks for your engagement and collaboration with the Firecracker Team and the rest of our members!
We achieved together the main milestone of Firecracker v1.0.0 which included new features, several improvements and we have just updated our public roadmap to track the effort of tasks in the near future, but we would like to hear your opinions!

We are inviting you to take part to a survey to rate your experience with Firecracker and its community between now and the 10th April  .
https://amazonmr.au1.qualtrics.com/jfe/form/SV_cDdtQT2m4pk2rf8

Please note that will take less than 10 minutes to complete the survey and all the responses are anonymous.
The aggregated results will be published as a newer discussion in the “General” category of our GitHub repo within the 29th April.

This is just the first step of a journey where we would like to establish a more solid and regular connection with the community and therefore understand how and when you would like to be engaged with us!

We are really obsessed with our customers and your experience it is important to us.
Our community grew substantially since the beginning of our journey and now, Firecracker is being used by many of you in different use cases  to solve different problems.

For the first time, we are taking the time in this survey to hear directly from you about your experience with Firecracker, covering topics as:

1. what do you like of Firecracker?
2. the challenges you faced to use it or contributing to it
3. which features or support you will need to solve your problem 

We can not wait to see your answers!
Thanks for your time :) 

Sincerely,
The Firecracker Team

**Comment by cipriancraciun** [comment]
Well, I've just completed the survey, and I found it a bit biased if not annoying in places...  I'll detail why.

First of all some context.  I don't use Firecracker (yet) in a professional capacity, I mainly use it for experimentation.  However I do keep an eye open on Firecracker developments, especially to see what new use-cases and opportunities it can enable.  (I do intend to use it in a startup, as a means to sandbox untrusted user code.)


Here is my feedback about the survey itself (not about the project):

* the survey is clearly biased towards IaaS / PaaS / SaaS providers that have already integrated Firecracker as part of their products and offerings;  however there might be use-cases outside of these, like for example for development and testing (by deploying lightweight VM's with Firecracker as opposed to Qemu or VirtualBox);

* in addition to the above, the survey doesn't make sense for those that just "watch" the Firecracker project, as opposed to actively using it;  however input from such people might be useful, because you could learn what would tip them into using Firecracker;  (perhaps you should mention that this survey applies only to those actively using the project;)

* there are some questions that I can't actually answer, as they don't apply to me, but there is no way to skip them;  (like for example how long it took to integrate it in my product;  I didn't integrate it yet, and there is no way to skip or say `N/A`, so I've chosen `less than 6 months` as I had no other option;  thus it will skew your results;  another example is the IOPS and bandwith requirements for which I don't yet have any, thus I've chosen the first option;)

* there is no support for going back to a previous answer;  (for example I didn't know what `MMDS` is, and on a subsequent question I've looked it up, and I've realized that "yes, I do want to use the meta-data service", but there was no going back;)


However I think it's a positive step in the right direction!

**Comment by xmarcalx** [comment]
Dear @cipriancraciun ,

Thank you very much for your detailed feedback, we will definitely take it in consideration for the next time.

Just a note for you and to help people who read this post and intend to take the survey as well:
- The majority of the questions in the survey (including the one you listed in your message) can be skipped just pressing the "forward arrow button" (->) because we actually considered that some users may not have an answer for them according the type of usage of Firecracker 
-  Very questions are mandatory so if you try to skip them it will ask you to answer. In such case, we always tried to cover all the use cases  including N/A or an "other" field according the type of question.
- We are trying to collect inputs from anyone, members who used Firecracker and members who are not using it and we encourage people to answer to all the questions listing features and new ideas that would make Firecracker a better tool for their own usecase and for the community itself!

Thanks again for your time to complete the survey and for your honest opinion!
